### PR TITLE
chore(makefile): Update makefile to push canary tag when building new image

### PR DESCRIPTION
**Pull request by jchauncey**

### CODE REVIEW

**Comment by vdice** [comment]
Wondering if we want to keep this process aligned with the current precedents in our other component repos, i.e. using a `versioning.mk` as seen in [controller](https://github.com/deis/controller/blob/master/versioning.mk)

This will prove to be useful in the future if we pull in something like [makeup](https://github.com/sgoings/makeup) where this logic (`versioning.mk` and additional) would exist in one canonical place and each deis component would apply said logic via this submodule.

File: Makefile [hunk 1]
```diff
@@ -49,6 +44,10 @@ dev: check-docker
 bootstrap: check-docker
 	${DEV_ENV_CMD} glide install
 
+# This is so you can build the binary without using docker
+build-binary:
+	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags ${LDFLAGS} -o $(BINARY_DEST_DIR)/logger main.go
```
Comment (arschles) [inline Makefile]: I'd rather have this build inside the `go-dev` docker container, as we do in other projects

File: versioning.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+.PHONY: docker-push
+docker-push: docker-mutable-push docker-immutable-push
+
+.PHONY: docker-immutable-push
+docker-immutable-push:
+	docker push ${IMAGE}
+
+.PHONY: docker-mutable-push
+docker-mutable-push:
+	docker push ${MUTABLE_IMAGE}
```
Comment (arschles) [inline versioning.mk]: nit: can you add a newline at the end?

Comment (jchauncey) [inline versioning.mk]: done

File: Makefile (hunk 1 above)
Comment (jchauncey) [inline Makefile]: There is a build with docker target already that we use for CI. The build-binary target is for local development where a person is not running docker-machine or boot2docker

**Pull request merged**